lms/lmstests/public/identical_tests/services.py

```diff
@@ -0,0 +1,96 @@
+import collections
+import logging
+
+from lms.lmsdb import models
+
+
+class IdenticalSolutionSolver:
+    def __init__(
+            self,
+            solution_check_pk: str,
+            logger: logging.Logger,
+    ):
+        self._solution_id = solution_check_pk
+        self._solution = None
+        self._logger = logger
+
+    def initialize(self):
+        self._solution = models.Solution.get_by_id(self._solution_id)
+
+    @property
+    def solution(self) -> models.Solution:
+        return self._solution
+
+    def check_identical(self):
+        for solution in models.Solution.select().join(
```

Comment:
It's better to split the check to another method we call `check_identical`

Reply:
Not sure I understand

Reply:
The function does 2 things (SRP violation): checks if something is identical and also clones the comment